Odata-docs/extensions/migration.md

```diff
@@ -0,0 +1,262 @@
+2) Utilizing the UriSegmentTranslator (`Middleware/UriSegmentTranslator.cs`) to convert each segment to an equivalent OData V4 segment
+3) Building the V4 path
+
+Translating each segment from V3 to V4 is a process of obtaining the relevant inner pieces of data of the V3 segment, looking up their equivalents
+in the V4 model, and constructing a new V4 segment with matching inner data.
+
+#### Query Translation
+There are only a few key differences between OData V3 and OData V4 when it comes to queries.  Here is the list of OData query options and their differences:
+
+|Query option|Description|Syntactic difference between V3 and V4|
+|---|---|---|
+|$expand|Expands related entities inline|None|
+|$filter|Filters the results, based on a Boolean condition|The same differences between URLs, for example the presence of keywords like "guid" and "datetime"|
+|$inlinecount|Tells the server to include the total count of matching entities in the response.|Changed in V4 to $count, either true or false (whereas $inlinecount is allpages or none)|
+|$orderby|Sorts the results|None|
+|$select|Selects which properties to include in the response|None|
+|$skip|Skips the first n results|None|
+|$top|Returns only the first n results|None|
+
+The OData Migration extension translates a V3 query by looking for either filter or inlinecount options and translating them.  The $filter option is translated by:
+
+1) Parsing the filter expression in OData V3
+2) Translating the OData v3 representation of the filter expression into an OData v4 representation of the filter expression
+3) Inserting the OData v4 representation back into the URL.
+
+The $inlinecount option is translated by changing "allpages" to "true" and "none" to "false".
+
+### Request Body Translation
+Request body translation is done by overridding the OData V4 ASP.NET Core InputFormatter.  InputFormatters in ASP.NET Core are responsible for deserializing incoming request bodies.
+In OData V4, the ODataInputFormatter uses a set of deserializers that are each responsible for a certain kind of request body payload.  For example, there is an ODataResourceDeserializer,
+which is responsible for deserializing incoming entities and complex types.  These deserializers are dispatched by the ODataDeserializerProvider; in OData V4 the implementation for the
+default provider is found in the `DefaultODataDeserializerProvider.cs` class.
+
+The main difference between the ODataMigrationInputFormatter and ODataInputFormatter is that the ODataMigrationInputFormatter uses a customized provider found in
+`ODataMigrationDeserializerProvider.cs`.  This is because OData Migration handles incoming OData V3 request bodies by dispatching customized deserializers.
+Each of these custom deserializers overrides the default OData V4 deserializer in order to insert logic that modifies the incoming request body _before_
+it is deserialized in OData V4.
+
+![](Images/RequestBody.PNG)
```

Comment:
Please add image alt tag. This will get flagged in our publish results.

Reply:
added